File: p283.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int num_zero = count(nums.begin(), nums.end(), 0);
        for(int i {0}; i < nums.size() - num_zero; ++i){
            if(nums[i] == 0){
                nums.push_back(0);
                nums.erase(nums.begin() + i);
                --i;
            }
        }
    }
};

File: p283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        idx = 0
        n = len(nums)

        while idx < n:
            if nums[idx] == 0:
                del nums[idx]
                nums.append(0)
                n -= 1
            else:
                idx += 1
